Flow.Launcher.Core/Plugin/PluginsLoader.cs

```diff
@@ -129,47 +133,103 @@ public static IEnumerable<PluginPair> PythonPlugins(List<PluginMetadata> source,
+                    Log.Error("PluginsLoader",$"Tried to automatically set from Settings.PythonDirectory " +
+                        $"but can't find python executable in {path}", "PythonPlugins");
                 }
             }
 
-            // if we have a path to the python executable,
-            // load every python plugin pair.
-            if (String.IsNullOrEmpty(Constant.PythonPath))
+            if (string.IsNullOrEmpty(settings.PythonDirectory))
             {
-                return new List<PluginPair>();
+                if (MessageBox.Show("Flow detected you have installed Python plugins, " +
+                        "would you like to install Python to run them? " +
+                        Environment.NewLine + Environment.NewLine +
+                        "Click no if it's already installed, " +
+                        "and you will be prompted to select the folder that contains the Python executable",
+                        string.Empty, MessageBoxButtons.YesNo) == DialogResult.No
+                    && string.IsNullOrEmpty(settings.PythonDirectory))
+                {
+                    var dlg = new FolderBrowserDialog
+                    {
+                        SelectedPath = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles)
+                    };
+
+                    var result = dlg.ShowDialog();
+                    if (result == DialogResult.OK)
+                    {
+                        string pythonDirectory = dlg.SelectedPath;
+                        if (!string.IsNullOrEmpty(pythonDirectory))
+                        {
+                            var pythonPath = Path.Combine(pythonDirectory, PythonExecutable);
+                            if (File.Exists(pythonPath))
+                            {
+                                settings.PythonDirectory = pythonDirectory;
+                                Constant.PythonPath = pythonPath;
+                            }
+                            else
+                            {
+                                MessageBox.Show("Can't find python in given directory");
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    DroplexPackage.Drop(App.python3_9_1).Wait();
+
+                    var installedPythonDirectory =
+                        Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), @"Programs\Python\Python39");
+                    var pythonPath = Path.Combine(installedPythonDirectory, PythonExecutable);
+                    if (FilesFolders.FileExists(pythonPath))
+                    {
+                        settings.PythonDirectory = installedPythonDirectory;
+                        Constant.PythonPath = pythonPath;
+                    }
+                    else
+                    {
+                        Log.Error("PluginsLoader", 
```

Comment:
What if user choose to install python on other path instead of the default path? I think we should allow selection if that it not found.

Reply:
this is to auto set path after auto install. Users can choose the path before hand

Reply:
Well I mean what if user don't choose to install python on the default path?

Reply:
Happy Chinese New Year @taooceros  :)

The installation will just install to the default location, it's a passive instead, no prompt for install location

Reply:
Happy New Year!!! 😃😃
Oh I see. That will be great!